js/web/lib/onnxjs/backends/webgl/program-manager.ts

```diff
@@ -69,15 +69,21 @@ export class ProgramManager {
     return this.profiler.event('backend', 'ProgramManager.build', () => {
       const preprocessor = new GlslPreprocessor(this.glContext, programInfo);
       const fragScript = preprocessor.preprocess();
-      const program = this.compile(fragScript);
-      const artifact = {
-        programInfo,
-        program,
-        uniformLocations: this.getUniformLocations(
-            program, preprocessor.context.programInfo.samplers, preprocessor.context.programInfo.variables),
-        attribLocations: this.getAttribLocations(program)
-      };
-      return artifact;
+      try {
+        const program = this.compile(fragScript);
+        const artifact = {
+          programInfo,
+          program,
+          uniformLocations: this.getUniformLocations(
+              program, preprocessor.context.programInfo.samplers, preprocessor.context.programInfo.variables),
+          attribLocations: this.getAttribLocations(program)
+        };
+        return artifact;
+
+      } catch (e) {
+        Logger.error('ProgramManager', fragScript);
```

Comment:
we should have already print the fragScript inside `compile()` (with env.debug === true).

imo the fragScript should not be print out to normal end users, which may only confuse them.

Reply:
This is only entered when compilation fails, i think most of the times it's for developers not normal end users. It's more convenient to have the shader code along with the compilation error displayed next to each other since the gl compiler doesnt return the faulty code block. Using debug flag gives you all the shader code from all kernels which actually makes debugging multi-kernel execution more difficult.

Reply:
This sounds reasonable.

I think the best place to print out the script source is here https://github.com/microsoft/onnxruntime/blob/master/js/web/lib/onnxjs/backends/webgl/webgl-context.ts#L178-L180